### PR TITLE
Fix for SetFit Auto

### PR DESCRIPTION
**Pull request by franz101**
SetFit auto was only working for train and eval, now added training arguments and support for inference and test

### CODE REVIEW

**Review by setu4993 — COMMENTED** [review]
Nice fix, looks good.

Can we add / update some tests here before we merge?

File: dataquality/integrations/setfit.py [hunk 1]
```diff
@@ -179,84 +182,87 @@ def auto(
...
+    Given a pandas dataframe, a file path, or a Huggingface dataset path, this
+    function will load the data, train a Huggingface transformer model, and
+    provide insights via a link to the Console.
+
+    At least one of `hf_data`, `train_data` should be provided. If neither of
+    those are, a demo dataset will be used for training.
+
+    Parameters
+    ----------
+    hf_data : Union[DatasetDict, str], optional
```
Comment (Ben-Epstein) [inline dataquality/integrations/setfit.py:205]: does this make sense for setfit? Shouldn't the model have to be a SetFitModel or SetFitTrainer?

File: dataquality/integrations/setfit.py [hunk 2]
```diff
@@ -308,19 +314,60 @@ def auto(
...
+            TaskType.text_classification, project_name=project_name, run_name=run_name
+        )
     dq.set_labels_for_run(labels)
-    _log_dataset_dict(dd)
-    trainer, encoded_data = get_trainer(
-        dd, labels, hf_model, max_padding_length, num_train_epochs
+    if not isinstance(hf_model, str):
+        return do_model_eval(hf_model, dd, wait, create_data_embs)
+
+    trainer, encoded_data = get_trainer(dd, hf_model, training_args)
```
Comment (Ben-Epstein) [inline dataquality/integrations/setfit.py:333]: given this line, how does the line above `hf_model: str = "sentence-transformers/paraphrase-mpnet-base-v2",`
make sense as the default? SHouldn't it have to be a SetFitTrainer or SetFitModel?

**Comment by franz101** [comment]
> Nice fix, looks good.
> 
> Can we add / update some tests here before we merge?

Added a test for DQ auto

**Review by Ben-Epstein — APPROVED** [review]

File: dataquality/integrations/setfit.py (hunk 1 above)
Comment (franz101) [reply dataquality/integrations/setfit.py:205]: Changed naming 👍

File: dataquality/integrations/setfit.py (hunk 2 above)
Comment (franz101) [reply dataquality/integrations/setfit.py:333]: If a string is provided we finetune the model

**Pull request merged**